packages/multichain-network-controller/src/MultichainNetworkController.ts

```diff
@@ -0,0 +1,195 @@
+import type { AccountsControllerSetSelectedAccountAction } from '@metamask/accounts-controller';
+import {
+  BaseController,
+  type ControllerGetStateAction,
+  type ControllerStateChangeEvent,
+  type RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import type {
+  NetworkStatus,
+  NetworkControllerGetNetworkConfigurationByNetworkClientId,
+  NetworkControllerSetActiveNetworkAction,
+  NetworkControllerGetStateAction,
+  NetworkControllerStateChangeEvent,
+} from '@metamask/network-controller';
+import type { Draft } from 'immer';
+
+import {
+  bitcoinCaip2ChainId,
+} from './constants';
+
+const controllerName = 'MultichainNetworkController';
+
+export type MultichainNetworkMetadata = {
+  features: string[];
+  status: NetworkStatus;
+};
+
+export type MultichainNetworkConfiguration = {
+  chainId: string; // Should be Caip2 type
+  name: string;
+  nativeCurrency: string; // Should be Caip19 type
+  blockExplorerUrls: string[];
+  defaultBlockExplorerUrlIndex?: number;
+  lastUpdated?: number;
+  isEvm?: false;
+};
+
+/**
+ * State used by the {@link MultichainNetworkController} to cache network configurations.
+ */
+export type MultichainNetworkControllerState = {
+  multichainNetworkConfigurationsByChainId: Record<
```

Comment:
Do `multichainNetworkConfigurationsByChainId` and `multichainNetworksMetadata` get used at the moment? If not, thoughts on introducing them in a future PR?

Reply:
`multichainNetworkConfigurationsByChainId ` this one is used, although `multichainNetworksMetadata` could be only introduced in a future PR!
That would most likely bring into discussion some other variables like `isEvm` in `multichainNetworkConfigurationsByChainId `

Reply:
Ah, I see now where `multichainNetworkConfigurationsByChainId` is used. We only use it to look up a property and we don't use the network configuration part of it, but maybe that is planned for a future PR? If so feel free to ignore, just seemed odd.